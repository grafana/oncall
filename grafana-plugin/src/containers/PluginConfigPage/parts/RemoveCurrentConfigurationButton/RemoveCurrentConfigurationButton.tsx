import React, { FC } from 'react';

import { Button } from '@grafana/ui';

import { WithConfirm } from 'components/WithConfirm/WithConfirm';

type Props = {
  disabled: boolean;
  onClick: () => void;
};

export const RemoveCurrentConfigurationButton: FC<Props> = ({ disabled, onClick }) => (
  <WithConfirm title="Are you sure to delete the plugin configuration?" confirmText="Remove">
    <Button variant="destructive" onClick={onClick} size="md" disabled={disabled}>
      Remove current configuration
    </Button>
  </WithConfirm>
);
