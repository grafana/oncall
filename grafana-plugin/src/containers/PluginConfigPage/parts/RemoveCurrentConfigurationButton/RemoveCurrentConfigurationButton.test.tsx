import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { RemoveCurrentConfigurationButton } from './RemoveCurrentConfigurationButton';

describe('RemoveCurrentConfigurationButton', () => {
  test('It renders properly when enabled', () => {
    const component = render(<RemoveCurrentConfigurationButton onClick={() => {}} disabled={false} />);
    expect(component.baseElement).toMatchSnapshot();
  });

  test('It renders properly when disabled', () => {
    const component = render(<RemoveCurrentConfigurationButton onClick={() => {}} disabled />);
    expect(component.baseElement).toMatchSnapshot();
  });

  test('It calls the onClick handler when clicked', async () => {
    const mockedOnClick = jest.fn();

    render(<RemoveCurrentConfigurationButton onClick={mockedOnClick} disabled={false} />);

    // click the button, which opens the modal
    await userEvent.click(screen.getByRole('button'));
    // click the confirm button within the modal, which actually triggers the callback
    await userEvent.click(screen.getByText('Remove'));

    expect(mockedOnClick).toHaveBeenCalledWith();
    expect(mockedOnClick).toHaveBeenCalledTimes(1);
  });
});
