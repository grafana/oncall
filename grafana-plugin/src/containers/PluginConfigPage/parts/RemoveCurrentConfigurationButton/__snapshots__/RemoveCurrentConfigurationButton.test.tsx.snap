// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveCurrentConfigurationButton It renders properly when disabled 1`] = `
<body>
  <div>
    <button
      class="css-ttl745-button"
      disabled=""
      type="button"
    >
      <span
        class="css-1riaxdn"
      >
        Remove current configuration
      </span>
    </button>
  </div>
</body>
`;

exports[`RemoveCurrentConfigurationButton It renders properly when enabled 1`] = `
<body>
  <div>
    <button
      class="css-ttl745-button"
      type="button"
    >
      <span
        class="css-1riaxdn"
      >
        Remove current configuration
      </span>
    </button>
  </div>
</body>
`;
